invenio_records_rest/ext.py

```diff
@@ -26,9 +26,66 @@
 
 from __future__ import absolute_import, print_function
 
+from werkzeug.utils import import_string, cached_property
+
 from .views import create_blueprint
 
 
+class _RecordRESTState(object):
+    """Record REST state."""
+
+    def __init__(self, app):
+        """Initialize state."""
+        self.app = app
+        self._read_permission_factory = None
+        self._create_permission_factory = None
+        self._update_permission_factory = None
+        self._delete_permission_factory = None
+
+    @cached_property
+    def read_permission_factory(self):
+        """Load default read permission factory."""
+        if self._read_permission_factory is None:
+            imp = self.app.config[
+                "RECORDS_REST_DEFAULT_READ_PERMISSION_FACTORY"
```

Comment:
Yes you are right. Can I fix that in the next pull request (search integration) so that I don't spend too much time rebasing?